core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -92,6 +94,16 @@ class KryoSerializer(conf: SparkConf)
       new KryoOutput(bufferSize, math.max(bufferSize, maxBufferSize))
     }
 
+  @transient
+  private lazy val factory: KryoFactory = new KryoFactory() {
+    override def create: Kryo = {
+      newKryo()
+    }
+  }
+
+  @transient
+  lazy val pool = new KryoPool.Builder(factory).softReferences.build
```

Comment:
`private`?

Reply:
This is used directly by KryoSerializerInstance